elements/rh-audio-player/test/rh-audio-player.spec.ts

```diff
@@ -0,0 +1,598 @@
+          expect(element?.playbackRate).to.equal(startrate + 0.2);
+        });
+      });
+    });
+
+    describe('full mode', function() {
+      let starttime: number;
+
+      beforeEach(setupForMode('full'));
+
+      beforeEach(function() {
+        starttime = element.currentTime;
+      });
+
+      it('has width', assertHasWidth);
+
+      describe('tabbable controls', function() {
+        beforeEach(waitForCanplaythrough);
+        beforeEach(seek(1));
+        describe('tab 1', function() {
+          beforeEach(tab);
+          it('has width', assertHasWidth);
+
+          it('focuses seek range', function() {
+            expect(element.shadowRoot?.activeElement).to.be.an.instanceof(RhRange).and.to.have.id('time');
+          });
+          describe('tab 2', function() {
+            beforeEach(tab);
+            it('focuses mute button', function() {
+              expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('Mute');
+            });
+            describe('tab 3', function() {
+              beforeEach(tab);
+              it('focuses volume slider', function() {
+                expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('Volume');
+              });
+              describe('tab 4', function() {
+                beforeEach(tab);
+                it('focuses speed select', function() {
+                  expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('Speed');
+                });
+                describe('tab 5', function() {
+                  beforeEach(tab);
+                  it('focuses rewind button', function() {
+                    expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('Rewind 15 seconds');
+                  });
+                  describe('tab 6', function() {
+                    beforeEach(tab);
+                    it('focuses play button', function() {
+                      expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('Play');
+                    });
+                    describe('tab 7', function() {
+                      beforeEach(tab);
+                      it('focuses advance button', function() {
+                        expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('Advance 15 seconds');
+                      });
+                      describe('tab 8', function() {
+                        beforeEach(tab);
+                        it('focuses menu button', function() {
+                          expect(element.shadowRoot?.activeElement?.getAttribute('aria-label')).to.equal('More options');
+                        });
+                        describe('tab 9', function() {
+                          beforeEach(tab);
+                          it('reaches the end of focusable elements', function() {
+                            expect(element.shadowRoot?.activeElement).to.be.null;
```

Comment:
@nikkimk this looks like a true failure. I was able to tab to the (invisible) first menu item from the menu button, without pressing space or enter

Reply:
you can paste this snippet into the console on the demo page to verify

```js
setInterval(() => console.log(document.querySelector('rh-audio-player')?.shadowRoot?.activeElement), 500);
```